src/daft-recordbatch/src/ops/window.rs

```diff
@@ -75,4 +75,54 @@ impl RecordBatch {
         // Union the original data with the window result
         self.union(&window_result)
     }
+
+    /// Simplified version of window_agg_sorted that works with pre-sorted single partitions
+    pub fn window_agg_sorted_partition(
+        &self,
+        to_agg: &AggExpr,
+        name: String,
+        group_by: &[ExprRef],
+    ) -> DaftResult<Self> {
+        if group_by.is_empty() {
+            return Err(DaftError::ValueError(
+                "Group by cannot be empty for window aggregation".into(),
+            ));
+        }
+
+        if matches!(to_agg, AggExpr::MapGroups { .. }) {
+            return Err(DaftError::ValueError(
+                "MapGroups not supported in window functions".into(),
+            ));
+        }
+
+        // Since this is a single partition, we can just evaluate the aggregation expression directly
+        let agg_result = self.eval_agg_expression(to_agg, None)?;
+        let window_col = agg_result.rename(&name);
+
+        // The aggregation result might be just a single value, so we need to broadcast it
+        // to match the length of the partition
+        let broadcast_result = if window_col.len() != self.len() {
+            window_col.broadcast(self.len())?
+        } else {
+            window_col
+        };
```

Comment:
The aggregation result should always be a single value, right? So no need for the if statement, just always broadcast